config/util.go

```diff
@@ -15,6 +15,37 @@ import (
 	"github.com/choria-io/go-choria/confkey"
 )
 
+// ProjectConfigurationFiles returns any configuration file in the specified directory and their parents directories.
+func ProjectConfigurationFiles(path string, err error) []string {
```

Comment:
hmm, this will search all the way up to `/`? should we somehow restrict this to only directories the user own? Or is going all the way up the intended behaviour? I am not really sure whats best - it's probably harmless but I'm weary of grabbing random stuff from all over the place.  Hard to anticipate all the implications and what kind of abuse a user other than you can do if they can influence your config?

Reply:
I also had concerns about this and finally decided that going to the filesystem root was maybe preferable: while usually a had practice, users might be working on project outside of their `$HOME` directory (e.g. from the deployment directory of an application, an NFS share or whatever) and in a Continuous Deployment context `$HOME` might not be set and the deployment user might be rooted in a file hierarchy that does not belong to them (I do have this case, but the CD system setup choria "system-wide" in the CD environment before triggering the deployment, so projects configuration files are not really an issue neither).

My assumption here is that the system is trustworthy and files which are not under the control of the user are as "safe" as the system-wide configuration file…

Reply:
@vjanelle what do you think?

Reply:
Doesn't match up well with the LSB,  but CI/CD use cases and containers throw all of that out of the window.

Whoever owns higher level directories most certainly has higher privileges - if this is the server component, that could be a bad thing, but for clients it's not an issue.

Reply:
File ownership could become a thing - I'd like to see this disabled by default ideally.